### PR TITLE
Skip samples that have a NaN value.

### LINKED ISSUE #152: Count values shoot off to infinity
Some of our metrics from our Kubernetes app sent to Stackdriver via stackdriver-prometheus-sidecar result in Stackdriver graphs which shoot off to infinity. Our alerts are triggered when this happens. Plots for the same time series look normal in Prometheus. I was wondering if anyone else is experiencing this, or could suggest what the cause was?

Selection_065
Selection_067
Selection_069

Thanks for any assistance you can give.

### PR DESCRIPTION
**Pull request by jkschulz**
Fixes #152 
Tested by running the script provided in issue #152 with image built with the fix.

Screen Shot 2019-10-08 at 9 26 35 PM

### CODE REVIEW

File: retrieval/transform_test.go [hunk 1]
```diff
@@ -760,6 +761,53 @@ func TestSampleBuilder(t *testing.T) {
...
+			},
+			metadata: metadataMap{
+				"job1/instance1/metric1": &scrape.MetricMetadata{Type: textparse.MetricTypeSummary, Metric: "metric1_count"},
+			},
+			metricPrefix: "test.googleapis.com",
+			input: []tsdb.RefSample{
+				// Need multiple samples, since the first will always be nil due to reset timestamp handling
+				{Ref: 1, T: 2000, V: 5},
+				{Ref: 1, T: 3000, V: 8},
+				{Ref: 1, T: 4000, V: math.NaN()},
```
Comment (StevenYCChou) [inline retrieval/transform_test.go]: Let's also test when a new value is added after NaN is recorded as this edge case may happen. For example:
```
				{Ref: 1, T: 2000, V: 5},
				{Ref: 1, T: 3000, V: 8},
				{Ref: 1, T: 4000, V: math.NaN()},
				{Ref: 1, T: 5000, V: 9},
```

File: retrieval/transform_test.go [hunk 2]
```diff
@@ -760,6 +761,121 @@ func TestSampleBuilder(t *testing.T) {
...
+						},
+						Value: &monitoring_pb.TypedValue{
+							Value: &monitoring_pb.TypedValue_Int64Value{3},
+						},
+					}},
+				},
+				nil,
+			},
+		},
+		{
```
Comment (StevenYCChou) [inline retrieval/transform_test.go]: Nit: Add comments in the beginning will help for readability, as this is a testcase with 60+ lines.

File: retrieval/transform_test.go [hunk 3]
```diff
@@ -760,6 +761,121 @@ func TestSampleBuilder(t *testing.T) {
...
+						Interval: &monitoring_pb.TimeInterval{
+							StartTime: &timestamp_pb.Timestamp{Seconds: 2},
+							EndTime:   &timestamp_pb.Timestamp{Seconds: 3},
+						},
+						Value: &monitoring_pb.TypedValue{
+							Value: &monitoring_pb.TypedValue_Int64Value{3},
+						},
+					}},
+				},
+				nil,
```
Comment (StevenYCChou) [inline retrieval/transform_test.go]: Nit: add a comment on this 2nd `nil` as this is the value we expect to see, and 2nd `nil` has different meaning compared to 1st `nil`.

File: retrieval/transform_test.go [hunk 4]
```diff
@@ -760,6 +761,121 @@ func TestSampleBuilder(t *testing.T) {
...
+			series: seriesMap{
+				1: labels.FromStrings("job", "job1", "instance", "instance1", "__name__", "metric1_count"),
+			},
+			metadata: metadataMap{
+				"job1/instance1/metric1": &scrape.MetricMetadata{Type: textparse.MetricTypeSummary, Metric: "metric1_count"},
+			},
+			metricPrefix: "test.googleapis.com",
+			input: []tsdb.RefSample{
+				// Need multiple samples, since the first will always be nil due to reset timestamp handling
+				{Ref: 1, T: 2000, V: 5},
```
Comment (StevenYCChou) [inline retrieval/transform_test.go]: Nit: As we are testing the `NaN` mainly, I think using 2 tsdb.RefSample as 
```
[]tsdb.RefSample{
				{Ref: 1, T: 2000, V: 5},
                                {Ref: 1, T: 4000, V: math.NaN()},
}
```
will help the reader understand this testcase more easily.

What do you think? If you agree, do not forget to apply this to the testcase next to this one.

Comment (jkschulz) [reply retrieval/transform_test.go]: Agreed.

File: retrieval/transform_test.go (hunk 2 above)
Comment (jkschulz) [reply retrieval/transform_test.go]: Done.

File: retrieval/transform_test.go (hunk 3 above)
Comment (jkschulz) [reply retrieval/transform_test.go]: Done.
